controllers/helpers.go

```diff
@@ -945,6 +945,10 @@ func MachinePoolToAzureManagedControlPlaneMapFunc(ctx context.Context, c client.
 
 		gk := gvk.GroupKind()
 		ref := cluster.Spec.ControlPlaneRef
+		if ref == nil || ref.Name == "" {
+			log.Error(err, "control plane ref is nil or empty")
```

Comment:
```suggestion
			log.Error(errors.New("control plane ref is nil or empty"))
```

I don't think you want to use `err` here as it will always be nil if it makes it past line 941.